datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -284,6 +288,7 @@ impl ExecutionPlan for ParquetExec {
             partition_index,
             projection: Arc::from(projection),
             batch_size: ctx.session_config().batch_size(),
+            predicate: self.predicate.clone(),
```

Comment:
I don't like all the cloning -- I have a follow on PR (https://github.com/apache/arrow-datafusion/pull/4283) to stop the copying